### PR TITLE
[retriever] Add ways to generate items from a PR

### PR DESCRIPTION
**Pull request by clee2000**
Like https://github.com/osalpekar/llm-target-determinator/pull/6 but without the multiple experiements/multiple pr parsing formats.

### CODE REVIEW

File: retriever.py [hunk 1]
```diff
@@ -73,72 +81,67 @@ def __init__(self, experiment_name):
...
+                        test = self.unittest_names[ind]
+                        score = similarity_matrix[ind]
+                        if test not in mapping:
+                            mapping[test] = []
+                        mapping[test].append(score.item())
+
+                # condense
+                for test, score in mapping.items():
+                    mapping[test] = sum(score) / len(score)
+                self.save_outputs(filename, mapping)
```
[the remark this answers is not in the document]
Comment (clee2000) [reply retriever.py:133]: There's a list of items, and each item gets it's own embedding + similarity matrix

File: retriever.py [hunk 2]
```diff
@@ -149,11 +152,25 @@ def main():
...
         help="Uses artifacts from the specified Indexer Experiment",
     )
+    parser.add_argument(
+        "--pr-parse-format",
+        type=str,
+        choices=PR_ITEMS.keys(),
+        required=True,
+        help="Specify what method to parse information from a PR",
+    )
+    parser.add_argument(
```
[the remark this answers is not in the document]
Comment (clee2000) [reply retriever.py:162]: Doesn't indexer take in experiment name as an input?

[the remark this answers is not in the document]
Comment (osalpekar) [reply retriever.py:162]: Yeah so basically the retriever output file can just be called experimentname-output.json or something like that

[the remark this answers is not in the document]
Comment (clee2000) [reply retriever.py]: setting as experimentname-prparseformat-output.json

**Pull request merged**